### PR TITLE
Feature/publisher

### PR DESCRIPTION
**Pull request by andrewmil**
Productionised the async code for full publisher service

### CODE REVIEW

**Review by warrenbailey — CHANGES_REQUESTED** [review]
I don't think we should include the Fake RAS endpoint and Dummy FTP server in the production service. They were just created for the spike.

What we could do is include them in SDX-compose to allow us to fully test this service.

Also have you tested this against the real RAS endpoint as they're were some changes we made to the GO spike we didn't make to the Python spike?

**Comment by warrenbailey** [comment]
The other thing we should add is a health check for the service, since we're using the aio libraries then we could reuse the code in sdx-rabbit-monitor

File: .trvais.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+language: python
```
Comment (tundish) [inline .trvais.yml:1]: Typo in Travis file name. Don't know how build is passing...

File: README.md [hunk 2]
```diff
@@ -1,26 +1,74 @@
...
+To run flake8 and the test scripts:
+
+    $ make test
 
 ## Configuration
 
-| Environment variable | Default | Description
-| -------------------- | ------- | -----------
-| PORT                 | -       | The port to bind to
+The following envioronment variables can be set:
```
Comment (tundish) [inline README.md:62]: Typo: envioronment

File: settings.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import logging
+import os
+
+
+LOGGING_LEVEL = logging.getLevelName(os.getenv('LOGGING_LEVEL', 'INFO'))
+
+FTP_HOST = os.getenv('FTP_HOST', "127.0.0.1")
```
Comment (tundish) [inline settings.py:7]: There is a mix of " and '. Pick just one ;-)

File: tests/test_publisher.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    async def test_publish_xlsx(self):
+        path = 'test.xlsx'
+        pub_rem = getattr(self.pub, 'publish_remove_files')
+        with asynctest.mock.patch("publisher.Publisher.publish") as mock_publish:
+            mock_ftp = asynctest.mock.Mock()
+            await pub_rem(path, ftp_client=mock_ftp, http_session=None)
+            mock_publish.assert_called_with()
+
+    async def test_not_publish_xlsx(self):
+        path = 'test.png'
```
Comment (tundish) [inline tests/test_publisher.py:46]: ' -> "

**Review by tundish — CHANGES_REQUESTED** [review]

**Pull request closed without merging**